airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -90,10 +90,13 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
         and returning an updated state object.
         """
-        latest_benchmark = latest_record[self.cursor_field]
-        if current_stream_state.get(self.cursor_field):
-            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
-        return {self.cursor_field: latest_benchmark}
+        updated_state = latest_record.get(self.cursor_field)
+        if updated_state:
+            stream_state_value = current_stream_state.get(self.cursor_field)
+            if stream_state_value:
+                updated_state = max(updated_state, stream_state_value)
+            current_stream_state[self.cursor_field] = updated_state
+        return current_stream_state
```

Comment:
According to spec.json `replication_start_date` is required and all cursor fields is date fields. So, we can used it as  default cursor value. Also please check request_params method (line 62). 

```suggestion
        updated_state = latest_record.get(self.cursor_field, self._replication_start_date)
        stream_state_value = current_stream_state.get(self.cursor_field, self._replication_start_date)
        current_stream_state[self.cursor_field] = max(updated_state, stream_state_value)
        
        return current_stream_state
```

Reply:
@lazebnyi
I have switched `replication_start_date` as default if `cursor field` is missed in record

about line 62
there is `if` condition because in "full refresh" streams `self._replication_start_date` still can be `None`.

Reply:
`replication_start_date` is required, so I don't see any situations where it can be empty during sync.

Reply:
we pass only to incremental stream like this 

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-pipedrive/source_pipedrive/source.py#L63

we don't pass to full_refresh streams like this:

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-pipedrive/source_pipedrive/source.py#L62